components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java

```diff
@@ -634,7 +635,7 @@ private void setClaimsToJwtClaimSet(Map<String, Object> claims, JWTClaimsSet jwt
                                 // prefix or in address scope.
                                 if (requestedClaims.contains(ADDRESS_PREFIX)) {
                                     claimsforAddressScope.put(entry.getKey().substring(ADDRESS_PREFIX.length()), claims.get(entry.getKey()));
-                                } else if (addressValues.contains(requestedClaims)) {
+                                } else if (addressValues != null && addressValues.contains(requestedClaims)) {
```

Comment:
IMO would be better to rename the 'addressValues' to something meaningful to reflect its meaning. Something in the line of addressScopeClaims.

'requestedClaims' could also be renamed to 'requestedClaim' since it only indicates a single claim whereas the current variable name is a bit misleading